subsys/net/lib/sockets/sockets.c

```diff
@@ -268,18 +287,23 @@ int _impl_zsock_listen(int sock, int backlog)
 #ifdef CONFIG_USERSPACE
 Z_SYSCALL_HANDLER(zsock_listen, sock, backlog)
 {
-	if (Z_SYSCALL_OBJ(sock, K_OBJ_NET_CONTEXT)) {
-		errno = EBADF;
-		return -1;
-	}
-
 	return _impl_zsock_listen(sock, backlog);
 }
 #endif /* CONFIG_USERSPACE */
 
 int _impl_zsock_accept(int sock, struct sockaddr *addr, socklen_t *addrlen)
 {
-	struct net_context *parent = INT_TO_POINTER(sock);
+	struct net_context *parent = sock_to_net_ctx(sock);
+	int fd;
+
+	if (parent == NULL) {
+		return -1;
+	}
+
+	fd = z_reserve_fd();
+	if (fd < 0) {
```

Comment:
z_free_fd(fd) needed here too, and other places where we violate MISRA Rule 15.5 (function should have only one exit)?

Reply:
Likewise, not needed. MISRA should not worry, in the compiler, all functions are turned into a CFG (control flow graph) with single exit. But source code is written for humans, and what can be easier to follow and prove then "if we can't continue, exit with error" structures? Clearly, not if/else straircases. The only better thing MISRA could suggest is a proper exception handling.